spyderlib/widgets/findreplace.py

```diff
@@ -198,6 +198,14 @@ def show(self):
         self.visibility_changed.emit(True)
         if self.editor is not None:
             text = self.editor.get_selected_text()
+
+            # If no text is highlighted for search, use whatever word is under the cursor
+            if not text:
+                cursor = self.editor.textCursor()
+                cursor.select(QTextCursor.WordUnderCursor)
+                text = to_text_string(cursor.selectedText())
+
+            # Now that text value is sorted out, use it for the search
             if len(text) > 0:

```

Comment:
`if text:` would be more pythonic. I know it's not from you but since you (correctly) used `if not text:` just before, it will be more coherent.